src/charm.py

```diff
@@ -122,21 +125,38 @@ def is_main(self) -> bool:
         """
         return self.get_main_unit() == self.unit.name
 
-    def peer_addresses(self) -> list:
-        """Property to get addresses of the units in the cluster.
+    def instance_map(self) -> typing.Dict:
+        """Build instance_map config.
 
         Returns:
-            List of peer IP addresses.
+            Instance map configuration as a dict.
         """
-        addresses = []
+        unit_name = self.unit.name.replace("/", "-")
+        app_name = self.app.name
+        model_name = self.model.name
+        addresses = [f"{unit_name}.{app_name}-endpoints.{model_name}.svc.cluster.local"]
         peer_relation = self.model.relations[synapse.SYNAPSE_PEER_RELATION_NAME]
         if peer_relation:
             relation = peer_relation[0]
             for u in relation.units:
-                if "ingress-address" in relation.data[u]:
-                    addresses.append(relation.data[u]["ingress-address"])
-        logger.debug("peer_addresses values are: %s", str(addresses))
-        return addresses
+                # <unit-name>.<app-name>-endpoints.<model-name>.svc.cluster.local
+                unit_name = u.name.replace("/", "-")
+                address = f"{unit_name}.{app_name}-endpoints.{model_name}.svc.cluster.local"
+                addresses.append(address)
+        logger.debug("addresses values are: %s", str(addresses))
+        instance_map = {}
+        main_unit_name = self.unit.name.replace("/", "-")
+        if self.get_main_unit() is not None and isinstance(self.get_main_unit, str):
+            main_unit_name = self.get_main_unit().replace("/", "-")
+        main_unit_address = f"{main_unit_name}.{app_name}-endpoints.{model_name}.svc.cluster.local"
+        for address in addresses:
+            match = re.search(r"-(\d+)", address)
+            if match is not None:
+                unit_number = match.group(1)
+            instance_name = "main" if address == main_unit_address else f"worker{unit_number}"
```

Comment:
Two lines before, there was a `if match is not None:`. If it was `None`, then `unit_number` is not defined. Can this happen?

Reply:
The check for match is none is more related to a lint issue than to the possibility of not matching the pattern. At least, this is what is described in the docs. 

"A unit is always named on the pattern <application>/<unit ID>, where <application> is the name of the application and the <unit ID> is its ID number or, for the leader unit, the keyword leader. For example, mysql/0 or mysql/leader. Note: the number designation is a static reference to a unique entity whereas the leader designation is a dynamic reference to whichever unit happens to be elected by Juju to be the leader ."

But I see that this check is confusing so I removed and added the comment with the docs. Thanks!